test/prow-cluster-operation/README.md

```diff
@@ -0,0 +1,57 @@
+## prow-cluster-operation
+
+prow-cluster-operation tool is a CLI tool for creating, deleting, getting a GKE
+cluster
+
+## Installation
+
+`go get knative.dev/test-infra/test/prow-cluster-operation`
+
+## Prerequisite
+
+- `GOOGLE_APPLICATION_CREDENTIALS` set
+
+## Usage
+
+This tool can be invoked from command line with following parameters:
+
+- `--min-nodes`: minumum number of nodes, default 1
+- `--max-nodes`: maximum number of nodes, default 3
+- `--node-type`: GCE node type, default "n1-standard-4"
+- `--region`: GKE region, default "us-central1"
+- `--zone`: GKE zone, default empty
+- `--project`" GCP project, default empty
+- `--name`: cluster name, default empty
+- `--backup-regions`: backup regions to be used if cluster creation in primary
+  region failed, comma separated list, default "us-west1,us-east1"
+- `--addons`: GKE addons, comma separated list, default empty
+
+## Flow
+
+### Create
+
+1. Acquiring cluster if kubeconfig already points to it
+1. Get GCP project name if not provided as a parameter:
+    - [In Prow] Acquire from Boskos
+    - [Not in Prow] Read from gcloud config
+
+    Failed obtaining project name will fail the tool
+1. Get default cluster name if not provided as a parameter
+1. Delete cluster if cluster with same name and location already exists in GKE
+1. Create cluster
+1. Write cluster metadata to `${ARTIFACT}/meta.json`
```

Comment:
Just curious, how is the file being used in the following steps?

Reply:
This is replacing https://github.com/knative/test-infra/blob/adedf32eed790736a3a1404a6c3e46d3fef8ad88/scripts/e2e-tests.sh#L138

Prow uses it for displaying metadata, and we also use it for setting up everything that's consumed by downstream cluster setup function